Social/Facebook/Oauth.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+/*
+ * This file is part of the Integrated package.
+ *
+ * (c) e-Active B.V. <integrated@e-active.nl>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Integrated\Bundle\SocialBundle\Social\Facebook;
+
+use Facebook\Exceptions\FacebookResponseException;
+use Facebook\Exceptions\FacebookSDKException;
+use Facebook\Facebook;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Class Oauth
+ * @package Integrated\Bundle\SocialBundle\Social\Facebook
+ */
+class Oauth
+{
+    /**
+     * @var ContainerInterface
+     */
+    private $container;
+
+    /**
+     * Oauth constructor.
+     * @param ContainerInterface $container
+     */
+    public function __construct(ContainerInterface $container)
+    {
+        $this->container = $container;
+    }
+
+    /**
+     * @param $connector
+     * @param $admin_url
+     * @return string
+     */
+    public function login($connector, $admin_url)
+    {
+        $fb = new Facebook([
+            'app_id' => $this->container->getParameter("app_id"),
+            'app_secret' => $this->container->getParameter("app_id_secret"),
+            'default_graph_version' => 'v2.2'
+        ]);
+
+        $helper = $fb->getRedirectLoginHelper();
+
+        $permissions = ['publish_actions']; // Optional permissions
+        $loginUrl = $helper->getLoginUrl(
+            "http://" . $_SERVER["SERVER_NAME"] . ":8080/" . $admin_url . '/connector/config/' . $connector,
```

Comment:
Could you please solve this url in an other way? See http://symfony.com/doc/current/components/http_foundation.html